enumerables.rb

```diff
@@ -0,0 +1,107 @@
+# Enumerable methods
+
+module Enumerable
+  # my_each
+
+  def my_each
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  # end of my_each
+
+  # my_each_with_index
+  def my_each_with_index
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  # end of my_each_with_index
+
+  # my_select
+
+  def my_select
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  # end of my_select
+
+  # my_all?
+
+  def my_all?
+    if_all = true
+    my_each do |i|
+      break if_all = false unless yield(i)
+    end
+    if_all
+  end
+
+  # end of my_all?
+
+  # my_any?
+
+  def my_any?
+    if_any = false
+    my_each do |i|
+      break if_any = true if yield(i)
+    end
+    if_any
+  end
+
+  # end of my_any?
+
+  # my_none?
+
+  def my_none?
+    if_none = true
+    my_each do |i|
+      break if_none = false if yield(i)
+    end
+    if_none
+  end
+
+  # end of my_none?
+
+  # my_count
+
+  def my_count(num)
+    count = 0
+    my_each { |i| count += 1 if num == i }
+    count
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81133334-857b8400-8f0e-11ea-8cfa-f41ec19907ad.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
[5, 4, 1].my_count { |x| x > 2 } == [5, 4, 1].count { |x| x > 2 }
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81133374-a93eca00-8f0e-11ea-9a04-6becb8c99851.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696440-055b1980-6f73-11ea-9b41-09dac295d3d3.png)